### PR TITLE
chore(docs): add design systems section to documentation

### LINKED ISSUE #3738: doc(design): re-work the design system documentation
At present, our documentation related to design systems is fairly adhoc. Over the last few weeks we've seen more and more community members coming to FAST who have had trouble figuring out how to accomplish design-system-related tasks. We should lift the "Design" docs out from under "Components" into a top level section in our ToC. Tentative name: "Working with Design Systems". We should plan out a more cohesive and comprehensive content strategy for this section which will walk our customers through the following three primary scenarios:

* Configuring the FAST Frame design system for use in an app
* Creating a new component that leverages a design system
* Creating a new design system from scratch

### PR DESCRIPTION
**Pull request by nicholasrice**
# Description
Documentation for FAST's Design System implementation and architecture is fragmented and hard to follow. This PR adds a section to the documentation to specifically address questions surrounding:
1. What the Design System is
2. How the Design System is used
3. How the Design System infrastructure functionally works
4. How to leverage the FAST Frame Design System
5. How to extend the existing FAST Frame Design System
6. How to create a new Design System

I wasn't sure what to do with the remaining content in the "Design" section under "Using FAST Components" - if any one has any ideas there please chime in.

closes #3738

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: sites/website/src/docs/design-systems/overview.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+id: overview
+title: What is the Design System?
+sidebar_label: Overview
+custom_edit_url: https://github.com/microsoft/fast/edit/master/sites/website/src/docs/design-systems/overview.md
+---
+A *Design System* is somewhat of an amorphous term but can generally be thought of as a collection of resources for interactive media that promotes brand alignment. While that definition is intentionally broad, in UI development, Design Systems generally manifest as component libraries surrounded by usage guidance and design principles.
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/overview.md:7]: What about... "A *Design System* is a collection of resources for interactive media that promotes brand alignment. While that definition is intentionally broad, in UI development, Design Systems manifest as component libraries surrounded by usage guidance and design principles."

File: sites/website/src/docs/design-systems/overview.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+In code, FAST currently uses "Design System" in a narrower sense than the above; the Design System is a dictionary of data that informs the visual (and sometimes functional) representation of UI elements. The Design System captures, stores, and reflects any information that should inform the rendering of a UI element. This data is often referred to as *Design Tokens*. Common examples of data in a Design System are:
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/overview.md:9]: What about... "FAST uses "Design System" in a narrower sense; the Design System is a dictionary of data that informs the visual (and sometimes functional) representation of UI elements."  

basically, I'm suggesting a more concise version.

File: sites/website/src/docs/design-systems/overview.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+- UI Colors
+- Fonts and type ramps
+- Motion data such as timings and easing curves
+- UI Density
+- Spacing values
+
+## How the Design System is used
+There are two primary ways that Design System values (Design Tokens) are used:
+1. In CSS stylesheets
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/overview.md:18]: Could it be "In CSS", because stylesheets is redundant and "In JavaScript".  Technically, both are programmatic, no?

File: sites/website/src/docs/design-systems/overview.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+In code, FAST currently uses "Design System" in a narrower sense than the above; the Design System is a dictionary of data that informs the visual (and sometimes functional) representation of UI elements. The Design System captures, stores, and reflects any information that should inform the rendering of a UI element. This data is often referred to as *Design Tokens*. Common examples of data in a Design System are:
+- UI Colors
+- Fonts and type ramps
+- Motion data such as timings and easing curves
+- UI Density
+- Spacing values
+
+## How the Design System is used
+There are two primary ways that Design System values (Design Tokens) are used:
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/overview.md:17]: I think we could eliminate "primary"

File: sites/website/src/docs/design-systems/overview.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+Many Design Tokens are used as CSS property values in component stylesheets. Assume for a moment that a Design Token `font-size-large` exists in a Design System, you may see something like the following:
+
+```css
+:host {
+    font-size: var(--font-size-large);
+}
+```
+
+### Design System in JavaScript
+It is sometimes the case that the Design System must be used in JavaScript. Assume for a moment that a Design Token `fadeInDuration` exists in a Design System, you may see something like the following:
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/overview.md:31]: a succinct way of saying the same thing may be:  "For Design Systems used in JavaScript, you may see a Design Token `fadeInDuration` shown as:" You could do the same above as well "For Design Systems used in CSS, you may see a Design Token `font-size-large` shown as:"

File: sites/website/src/docs/design-systems/overview.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+)
+```
+
+## The DesignSystemProvider
+The Design System itself manifests through a [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider): it is the vessel through which the Design System is expressed. The [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) is an HTML element that facilitates usage, configuration, and propagation of the Design System through a UI view. The [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) is responsible for expressing the Design System both as a readable JavaScript property *and* as CSS custom properties.
+
+### Design System flow
+It is important to understand that in FAST, the Design System is a mutable and inherited. You can think of the Design System as a flow of data through the DOM toward leaf nodes, where every [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) element inherits Design System data from it's closest ancestor [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) and provides opportunity to *change* that data for all descendent elements. Let's visualize this and assume there is a `font-size-large` Design Token in the Design System:
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/overview.md:45]: Omit "It is important to understand that" and could start with "The FAST Design System is a mutable..."

Comment (awentzel) [reply sites/website/src/docs/design-systems/overview.md:45]: omit "You can think of the" and start with "Think of the Design System..."

File: sites/website/src/docs/design-systems/overview.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+    <my-design-system-provider font-size-large="15px">
+        <my-text style="font-size: var(font-size-large);" id="two">
+            My font size is 15px
+        </my-text>
+    </my-design-system-provider>
+</my-design-system-provider>
+```
+
+As you can see above, the `font-size-large` Design Token is set on each [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) instance. The [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) reflects that value to a CSS custom property (more on that later), which the `<my-text>` element in it's stylesheet.
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/overview.md:62]: "As shown above,..."

File: sites/website/src/docs/design-systems/using-the-design-system.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
+---
+id: using-the-design-system
+title: Using the FAST Design System 
+sidebar_label: Using the Design System
+custom_edit_url: https://github.com/microsoft/fast/edit/master/sites/website/src/docs/design-systems/using-the-design-system.md
+---
+
+This section goes into further detail on how the Design System can be used. It will use [FAST Frame](/docs/design-systems/fast-frame) Design System in the examples but none are *specific* to [FAST Frame](/docs/design-systems/fast-frame); use these features with your own Design System if you're not using [FAST Frame](/docs/design-systems/fast-frame).
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/using-the-design-system.md:8]: I feel like the first sentence is given already in the title.  Would suggest removing

Comment (awentzel) [reply sites/website/src/docs/design-systems/using-the-design-system.md:8]: Maybe could begin with "Using FAST Frame in the Design system" or some variation perhaps

File: sites/website/src/docs/design-systems/using-the-design-system.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+### Setting Design System Properties
+Design System properties can always be set by simple property assignment in JavaScript. Simply obtain a reference to the Custom Element instance and assign the property.
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/using-the-design-system.md:11]: omit "simple"

Comment (awentzel) [reply sites/website/src/docs/design-systems/using-the-design-system.md:11]: "Design System properties can be set by property assignment in JavaScript by obtaining a reference to the Custom Element instance and assign the property."

File: sites/website/src/docs/design-systems/using-the-design-system.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+```html
+<fast-design-system-provider design-unit="6">
+
+</fast-design-system-provider>
+```
+
+## `use-defaults`
+Every [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) supports the `use-defaults` boolean attribute. `use-defaults` is a mechanism to assign the [default Design System values](/docs/design-systems/creating-a-design-system#default) to the [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider).
+
+It is generally a good idea to have a [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) with the `use-defaults` attribute wrapping all of the app UI; this will ensure all properties used by descendent components are defined. This is especially important if the Design System properties reflect to CSS custom properties because the CSS custom properties are only created when the value is set.
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/using-the-design-system.md:42]: omit "generally"

Comment (awentzel) [reply sites/website/src/docs/design-systems/using-the-design-system.md:42]: replace "wrapping all of the app UI; this will ensure" with "wrapping all of the app UI to ensure"

File: sites/website/src/docs/design-systems/using-the-design-system.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+
+**Example: `use-defaults` with attribute assignment**
+```html
+<fast-design-system-provider use-defaults design-unit="6">
+    <!-- the Design System's design unit will be 6, all other Design System values will be initialized to their default -->
+</fast-design-system-provider>
+```
+
+### Composing Design System Providers
+Remember from the [overview](/docs/design-systems/overview#design-system-flow) that the Design System values are inherited and propagate  down the DOM tree. This can be used to scope regions of a page with Design System changes.
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/using-the-design-system.md:69]: double spaces after "propagate"

File: sites/website/src/docs/design-systems/creating-a-design-system.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
+---
+id: creating-a-design-system
+title: Creating a Design System
+sidebar_label: Creating a Design System
+custom_edit_url: https://github.com/microsoft/fast/edit/master/sites/website/src/docs/design-systems/creating-a-design-system.md
+---
+Creating a design system involves two key steps:
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/creating-a-design-system.md:7]: omit "key"

File: sites/website/src/docs/design-systems/creating-a-design-system.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+1. Determine what information the Design System should hold
+2. Creating a [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) to communicate the Design System
+
+The information the Design System should hold will be dictated by a project's design philosophies and capabilities and can vary widely from project to project. Therefore, this documentation will focus on creating and configuring a [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) and not data in the Design System.
```
Comment (awentzel) [inline sites/website/src/docs/design-systems/creating-a-design-system.md:11]: "The Design System holds information dictated by a project's design guidelines, philosophies, and capabilities which may vary widely across projects."

File: sites/website/src/docs/design-systems/overview.md (hunk 5 above)
Comment (nicholasrice) [reply sites/website/src/docs/design-systems/overview.md:31]: Adjusting to be more succinct

File: sites/website/src/docs/design-systems/using-the-design-system.md (hunk 8 above)
Comment (nicholasrice) [reply sites/website/src/docs/design-systems/using-the-design-system.md:8]: I added this to be sure folks didn't conflate this with documentation that is specific to FAST Frame because it is general purpose documentation. I think removing this would cause confusion between using the FAST Frame Design System and using *any* Design System

File: sites/website/src/docs/design-systems/overview.md (hunk 1 above)
Comment (nicholasrice) [reply sites/website/src/docs/design-systems/overview.md:7]: The point is that there is a very broad definition in the industry about what a design system actually is - I'm not sure we want to declare the industry term here, just set the stage for how the term is used in FAST today

File: sites/website/src/docs/design-systems/overview.md [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+### Design System flow
+The Design System is a mutable and inherited. Think of the Design System as a flow of data through the DOM toward leaf nodes, where every [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) element inherits Design System data from it's closest ancestor [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) and provides opportunity to *change* that data for all descendent elements. Let's visualize this, and assume there is a `font-size-large` Design Token in the Design System:
+
+**Example: The Design System flows data down the DOM hierarchy**
+```html
+<my-design-system-provider font-size-large="28px">
+    <my-text style="font-size: var(font-size-large);" id="one">
```
Comment (EisenbergEffect) [inline sites/website/src/docs/design-systems/overview.md:50]:
```suggestion
    <my-text style="font-size: var(--font-size-large);" id="one">
```

File: sites/website/src/docs/design-systems/overview.md [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+        My font size is 29px
+    </my-text>
+
+    <my-design-system-provider font-size-large="15px">
+        <my-text style="font-size: var(font-size-large);" id="two">
```
Comment (EisenbergEffect) [inline sites/website/src/docs/design-systems/overview.md:55]:
```suggestion
        <my-text style="font-size: var(--font-size-large);" id="two">
```

File: sites/website/src/docs/design-systems/overview.md [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+    {duration: designSystem.fadeInDuration}
+)
+```
+
+## The DesignSystemProvider
+The Design System itself manifests through a [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider): it is the vessel through which the Design System is expressed. The [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) is an HTML element that facilitates usage, configuration, and propagation of the Design System through a UI view. The [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) is responsible for expressing the Design System both as a readable JavaScript property *and* as CSS custom properties.
+
+### Design System flow
+The Design System is a mutable and inherited. Think of the Design System as a flow of data through the DOM toward leaf nodes, where every [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) element inherits Design System data from it's closest ancestor [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) and provides opportunity to *change* that data for all descendent elements. Let's visualize this, and assume there is a `font-size-large` Design Token in the Design System:
```
Comment (EisenbergEffect) [inline sites/website/src/docs/design-systems/overview.md:45]:
```suggestion
The Design System is mutable and inherited. Think of the Design System as a flow of data through the DOM toward leaf nodes, where every [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) element inherits Design System data from it's closest ancestor [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) and provides opportunity to *change* that data for all descendent elements. Let's visualize this, and assume there is a `font-size-large` Design Token in the Design System:
```

File: sites/website/src/docs/design-systems/overview.md [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+
+    <my-design-system-provider font-size-large="15px">
+        <my-text style="font-size: var(font-size-large);" id="two">
+            My font size is 15px
+        </my-text>
+    </my-design-system-provider>
+</my-design-system-provider>
+```
+
+As shown above, the `font-size-large` Design System property is set on each [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) instance. The [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) reflects that value to a CSS custom property ([more on that later](/docs/design-systems/using-the-design-system#css-custom-properties)), which the `<my-text>` element in it's stylesheet.
```
Comment (EisenbergEffect) [inline sites/website/src/docs/design-systems/overview.md:62]: The last sentence seems incomplete.

File: sites/website/src/docs/design-systems/using-the-design-system.md [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+**Example: Register a CSS custom property that is a function of the Design System**
+```ts
+DesignSystemProvider.registerCSSCustomProperty({
+    name: "design-unit-to-the-12th",
+    value: designSystem => Math.pow(designSystem.designUnit, 12),
+});
+```
+
+#### CSSCustomPropertyBehavior
+The above API is especially useful when defining a [component stylesheet](docs/fast-element/leveraging-css); individual stylesheets can declare dependencies on CSS custom properties that are functions of the element instance's Design System through with the [CSSCustomPropertyBehavior](/docs/api/fast-foundation.csscustompropertybehavior). This is how the [FAST Frame Color Recipes](/docs/design-systems/fast-frame#color-recipes) work are constructed.
```
Comment (EisenbergEffect) [inline sites/website/src/docs/design-systems/using-the-design-system.md:110]:
```suggestion
The above API is especially useful when defining a [component stylesheet](docs/fast-element/leveraging-css); individual stylesheets can declare dependencies on CSS custom properties that are functions of the element instance's Design System through with the [CSSCustomPropertyBehavior](/docs/api/fast-foundation.csscustompropertybehavior). This is how the [FAST Frame Color Recipes](/docs/design-systems/fast-frame#color-recipes) are constructed.
```

File: sites/website/src/docs/design-systems/creating-a-design-system.md [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+**Example: Setting the property from the HTML attribute**
+```html
+<my-design-system-provider font-size-large="18px"></my-design-system-provider>
+```
+
+### `cssCustomProperty`
+The `cssCustomProperty` property is used to define the name of the [CSS custom property](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) to which the value is mapped. This allows easy use of the Design System value in CSS stylesheets. This property is optional and defaults to the attribute name of the decorated property; it can also be set to `false` to omit the [CSS custom property](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) association. In this case, the `cssCustomProperty` property is being used to spinal-case the property name.
+
+```ts
+const styles = css`:host { font-size: var(font-size-large)}`;
```
Comment (EisenbergEffect) [inline sites/website/src/docs/design-systems/creating-a-design-system.md:93]:
```suggestion
const styles = css`:host { font-size: var(--font-size-large)}`;
```

File: sites/website/src/docs/design-systems/fast-frame.md [hunk 20]
```diff
@@ -1,35 +1,68 @@
...
 
-## Algorithmic colors (recipes)
+Each Design System property in the [`FASTDesignSystemProvider`](/docs/api/fast-components.fastdesignsystemprovider) will list the HTML attribute on the `<fast-design-system-provider>` that controls the property, and also whether it creates a corresponding CSS custom property. For more information on how to *set* each Design System property see [Setting Design System Properties](/docs/design-systems/using-the-design-system#setting-design-system-properties), and for information on how to use Design System CSS custom properties see [Using Design System CSS Custom Properties](/docs/design-systems/using-the-design-system#css-custom-properties).
 
-FAST makes heavy use of algorithmic colors; named colors are a product of the *designSystem* object in which they are calculated. In the documentation below, these algorithmic colors will be referred to as *recipes*.
+## Using the FAST Frame Design System
+To use the FAST Frame Design System, simply ensure the [`FASTDesignSystemProvider`](/docs/api/fast-components.fastdesignsystemprovider) is define and use the element in your view:
```
Comment (EisenbergEffect) [inline sites/website/src/docs/design-systems/fast-frame.md:15]:
```suggestion
To use the FAST Frame Design System, simply ensure the [`FASTDesignSystemProvider`](/docs/api/fast-components.fastdesignsystemprovider) is defined and use the element in your view:
```

File: sites/website/src/docs/design-systems/fast-frame.md [hunk 21]
```diff
@@ -1,35 +1,68 @@
...
+FASTDesignSystemProvider; // Prevent tree shaking
+```
+```html
+<fast-design-system-provider use-defaults>
+    Hello world
+</fast-design-system-provider>
+```
+
+:::note
+It is encouraged to add the `<fast-design-system-provider>` at the root of your view with the `use-defaults` attribute. This ensures that all Design System properties are initialized. For more info see the [`use-defaults`](/docs/design-systems/using-the-design-system#use-defaults)
```
Comment (EisenbergEffect) [inline sites/website/src/docs/design-systems/fast-frame.md:30]:
```suggestion
We recommend to add the `<fast-design-system-provider>` at the root of your view with the `use-defaults` attribute. This ensures that all Design System properties are initialized. For more info see the [`use-defaults`](/docs/design-systems/using-the-design-system#use-defaults)
```

File: sites/website/src/docs/design-systems/overview.md (hunk 14 above)
Comment (nicholasrice) [reply sites/website/src/docs/design-systems/overview.md:50]: Good catch - thanks

File: sites/website/src/docs/design-systems/overview.md (hunk 17 above)
Comment (nicholasrice) [inline sites/website/src/docs/design-systems/overview.md:62]:
```suggestion
As shown above, the `font-size-large` Design System property is set on each [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) instance. The [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) reflects that value to a CSS custom property ([more on that later](/docs/design-systems/using-the-design-system#css-custom-properties)), which the `<my-text>` element uses in it's stylesheet.
```

File: sites/website/src/docs/design-systems/overview.md [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+        <my-text style="font-size: var(--font-size-large);" id="two">
+            My font size is 15px
+        </my-text>
+    </my-design-system-provider>
+</my-design-system-provider>
+```
+
+As shown above, the `font-size-large` Design System property is set on each [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) instance. The [DesignSystemProvider](/docs/api/fast-foundation.designsystemprovider) reflects that value to a CSS custom property ([more on that later](/docs/design-systems/using-the-design-system#css-custom-properties)), which the `<my-text>` element uses in it's stylesheet.
+
+But we can *also* access that hierarchial information from JavaScript, which opens up many advanced scenarios that will be explored in the [FAST Frame Design System](/docs/design-systems/fast-frame).
```
Comment (janechu) [inline sites/website/src/docs/design-systems/overview.md:64]:
```suggestion
But we can *also* access that hierarchical information from JavaScript, which opens up many advanced scenarios that will be explored in the [FAST Frame Design System](/docs/design-systems/fast-frame).
```

File: sites/website/src/docs/design-systems/fast-frame.md [hunk 23]
```diff
@@ -62,31 +94,29 @@ provider.accentBaseColor = accent;
...
+#### Offsets / Deltas
+If you look at the properties of the [`FASTDesignSystem`](/docs/api/fast-components.fastdesignsystem), you'll notice a number of "Delta" properties. These are used by color recipes to *shift* the colors derived from recipes by index of the palette. You can adjust these values to fine-tune each individual color recipe.
 
-## Color recipes
+#### Color recipes
+Color recipes are algorithmic colors generated relative to a context. Recipes can be thought of and used as named colors, however they are more than simple variables; they are generated values where the value of the Recipe will change depending on the context in which it is generated. The primary inputs to any color recipe are [Palettes](/docs/design-systems/fast-frame#palettes), [the background color](/docs/design-systems/fast-frame#background-color), and [offsets / deltas](/docs/design-systems/fast-frame#offsets-deltas), which come from the nearest ancestor `<fast-design-system-provider>`.
```
Comment (janechu) [inline sites/website/src/docs/design-systems/fast-frame.md:106]:
```suggestion
Color recipes are algorithmic colors generated relative to a context. Recipes can be thought of and used as named colors, however they are more than simple variables; they are generated values where the value of the Recipe will change depending on the context in which it is generated. The primary inputs to any color recipe are [Palettes](/docs/design-systems/fast-frame#palettes), the [background color](/docs/design-systems/fast-frame#background-color), and [offsets / deltas](/docs/design-systems/fast-frame#offsets-deltas), which come from the nearest ancestor `<fast-design-system-provider>`.
```

File: packages/web-components/fast-components/src/styles/recipes.ts [hunk 24]
```diff
@@ -337,6 +337,15 @@ export const neutralFillInputActiveBehavior = cssCustomPropertyBehaviorFactory(
     x => neutralFillInput(x).active,
     FASTDesignSystemProvider.findProvider
 );
+/**
+ * Behavior to resolve and make available the neutral-fill-input-selected CSS custom property.
+ * @public
+ */
+export const neutralFillInputSelectedBehavior = cssCustomPropertyBehaviorFactory(
```
[the remark this answers is not in the document]
Comment (nicholasrice) [reply packages/web-components/fast-components/src/styles/recipes.ts:344]: good catch

File: packages/web-components/fast-components/src/styles/recipes.ts [hunk 25]
```diff
@@ -337,6 +337,15 @@ export const neutralFillInputActiveBehavior = cssCustomPropertyBehaviorFactory(
...
+    "neutral-fill-input-active",
```
Comment (nicholasrice) [inline packages/web-components/fast-components/src/styles/recipes.ts:345]:
```suggestion
    "neutral-fill-input-selected",
```
